### DISCUSSION TITLE
Integrate Keycloak and OpenStack with Openid Connect

### DISCUSSION THREAD

**Discussion by C9504** [keycloak/keycloak#13277] [Q&A]
Hello everyone, does anyone know how to integrate Keycloak with openstack through Openid Connect?
If you know any documentation I would appreciate your collaboration.

**Comment by morganchenjp** [comment]
+1. 
alo want to integrate Keystone w/ Keycloak, and share same token generated by KeyCloak.

**Comment by srini4174** [comment]
> Hello everyone, does anyone know how to integrate Keycloak with openstack through Openid Connect? If you know any documentation I would appreciate your collaboration.

Hi @C9504 and @morganchenjp Have you found a solution for integrating OpenStack with Keycloak via OpenID Connect? If yes means please help me to do the same. Thank you.

**Reply by morganchenjp** [reply]
finally, out last choice is that write a plugin of Keycloak, subscribe the user change event inside keycloak,  then the plugin will sync change to Keystone.
